src/utilities/index.ts

```diff
@@ -189,5 +190,7 @@ export const mergeErrors = <T extends FieldValues>(
     }
   }
 
+  onMerge && onMerge();
```

Comment:
maybe I am missing something but wouldn't this trigger a re-render right after merging errors and remove the invalid state right away? I think the real solution is to allow you to remove the errors when you want via a reset function instead.

Reply:
This should not be our decision to make I think, I would augment the reset form handler only, so instead of it reseting the form only it also sets the flag to false, and the be errors will be passed and merged only when the flag is true (which happens from the moment you successfully submit to the moment you reset the form)

Reply:
`you reset the form `

What do you mean by the above? Do you mean call reset?

Note, The above configuration restores the expected behavior of react-hook-form. I agree that we should add a method to allow the user to restore the default behavior of remix-run useActionData though, so how about we add restorePersistedActionData: boolean;